hazelcast/src/main/java/com/hazelcast/map/impl/recordstore/expiry/ExpirySystem.java

```diff
@@ -200,11 +227,26 @@ public final void extendExpiryTime(Data dataKey, long now) {
         }
 
         ExpiryMetadata expiryMetadata = getExpiryMetadataForExpiryCheck(dataKey, expireTimeByKey);
-        if (expiryMetadata == null
-                || expiryMetadata.getMaxIdle() == Long.MAX_VALUE) {
+        if (expiryMetadata == null) {
             return;
         }
 
+        long maxIdle = expiryMetadata.getMaxIdle();
+        if (maxIdle == Long.MAX_VALUE) {
+            return;
+        }
+
+        long ttl = expiryMetadata.getTtl();
+        if (ttl <= maxIdle) {
+            return;
+        }
+
+        // RU_COMPAT_4_2
+        if (isClusterVersionLessThanV5()) {
+            checkIfTtlGreaterThanMaxIdle(recordStore.getName(),
+                    lastUpdateTime, maxIdle, ttl);
+        }
```

Comment:
I think it is redundant though I'm not so sure: see my comment to the 4.2 PR: https://github.com/hazelcast/hazelcast/pull/19483#discussion_r707978953